org.eclipse.buildship.core.test/src/main/groovy/org/eclipse/buildship/core/workspace/internal/SingleProjectSynchronizationSpecification.groovy

```diff
@@ -270,16 +327,28 @@ abstract class SingleProjectSynchronizationSpecification extends ProjectSynchron
         }
 
         when:
-        synchronizeAndWait(projectDir)
+        importAndWait(projectDir, distribution)
+        String outputLocation = findJavaProject('sample-project').outputLocation?.toPortableString()
 
         then:
-        findJavaProject('sample-project').getOutputLocation().toPortableString() == '/sample-project/target/bin'
+        if (higherOrEqual('3.0', distribution)) {
```

Comment:
All these if-statements make the tests pretty hard to follow. You might want to take the approach from Gradle core instead where we have a `@TargetGradleVersion` annotation on each test method, so we can split the different cases into different methods.

Reply:
I'll try to copy the test fixtures.

Reply:
I tried to reuse `@TargetGradleVersion`. Copying over the custom test runner adds a lot of extra code which seems to be too much of an extra maintenance burden. I can still split the tests based on the version requirement.

Reply:
I've refactored the tests not to have those `higherOrEqual` conditions in the test assertions.